src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -46,6 +46,8 @@ public ref partial struct Utf8JsonReader
         private SequencePosition _currentPosition;
         private readonly ReadOnlySequence<byte> _sequence;
 
+        internal bool _hasPartialStringValue;
```

Comment:
Like with Utf8JsonWriter, couldn't this just use the same special `JsonTokenType` value to reflect segmented reads?